tests/src/core/testqgsfield.cpp

```diff
@@ -336,6 +351,13 @@ void TestQgsField::equality()
   field2.setSplitPolicy( Qgis::FieldDomainSplitPolicy::GeometryRatio );
   QVERIFY( field1 == field2 );
 
+  field1.setMetadata( {{ static_cast< int >( Qgis::FieldMetadataProperty::GeometryCrs ), QStringLiteral( "abc" )}, {2, 5 }} );
+  QVERIFY( !( field1 == field2 ) );
+  QVERIFY( field1 != field2 );
```

Comment:
Should these tests be put in a metadata-specific test rather than in ::isNumeric ?

Reply:
See above -- this is in the equality test